modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -451,25 +465,28 @@ class ConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
     {
 #ifdef HAVE_INF_ENGINE
         InferenceEngine::DataPtr input = infEngineDataNode(inputs[0]);
-        CV_Assert(input->dims.size() == 4);
+        CV_Assert(input->dims.size() == 4 || input->dims.size() == 5);
 
-        const int inpCn = input->dims[2];  // NOTE: input->dims are reversed (whcn)
+        const int inpCn = input->dims[input->dims.size() - 2];  // NOTE: input->dims are reversed (whcn or whdcn)
```

Comment:
Please rename `whcn` to `WHIO` (and `whdcn` to `WHDIO` correspondingly) due there is no batch (n) and channels (c) in the kernel but there are input channels (I) and output channels (O).